### PR TITLE
feat(descriptions): add new feature for customized style

### PR DESCRIPTION
**Pull request by kooriookami**
add props: width, min-width, align, label-align, class-name, label-class-name

re #1976 #2353

Please make sure these boxes are checked before submitting your PR, thank you!

* [ ] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [ ] Make sure you are merging your commits to `dev` branch.
* [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/descriptions/src/descriptions-cell.ts [hunk 1]
```diff
@@ -16,39 +17,73 @@ export default defineComponent({
   },
   setup(props) {
     const descriptions = inject(elDescriptionsKey, {} as IDescriptionsInject)
+    const item = computed(() => {
+      return {
```
Comment (HerringtonDarkholme) [inline packages/descriptions/src/descriptions-cell.ts:21]: Can we extract out pros.cell.props?

File: packages/descriptions/src/descriptions-cell.ts [hunk 2]
```diff
@@ -16,39 +17,73 @@ export default defineComponent({
...
+        label: props.cell?.props?.label || '',
+        span: props.cell?.props?.span || 1,
+        width: props.cell?.props?.width || '',
+        minWidth: props.cell?.props?.['min-width'] || props.cell?.props?.minWidth || '',
+        align: props.cell?.props?.align || '',
+        labelAlign: props.cell?.props?.['label-align'] || props.cell?.props?.labelAlign || '',
```
Comment (HerringtonDarkholme) [inline packages/descriptions/src/descriptions-cell.ts:27]: Does Vue help us to normalize prop name?

File: packages/descriptions/src/descriptions-cell.ts [hunk 3]
```diff
@@ -16,39 +17,73 @@ export default defineComponent({
...
+      span,
+      align,
+      labelAlign,
+      className,
+      labelClassName,
+      style,
     }
   },
   render() {
     switch (this.type) {
```
Comment (HerringtonDarkholme) [inline packages/descriptions/src/descriptions-cell.ts:63]: Looks like template fits better in rendering simple logic like this.

**Review by HerringtonDarkholme — CHANGES_REQUESTED** [review]
Comments added

File: packages/descriptions/src/descriptions-cell.ts (hunk 2 above)
Comment (kooriookami) [reply packages/descriptions/src/descriptions-cell.ts:27]: How to do?

File: packages/descriptions/src/descriptions-cell.ts [hunk 4]
```diff
@@ -1,5 +1,7 @@
-import { computed, defineComponent, h, inject } from 'vue'
-import { elDescriptionsKey, IDescriptionsInject } from './descriptions.type'
+import { defineComponent, h, inject, VNode } from 'vue'
+import { elDescriptionsKey, IDescriptionsInject, IDescriptionsItemInject } from './descriptions.type'
```
Comment (JeremyWuuuuu) [inline packages/descriptions/src/descriptions-cell.ts:2]: Separate `type` and actual `variable` and mark `type` as `type`.

**Comment by JeremyWuuuuu** [comment]
I wonder why is this have to do with `table` documentations, did you somehow mixed your branches together?

**Comment by kooriookami** [comment]
> I wonder why is this have to do with `table` documentations, did you somehow mixed your branches together?

Some props are silimar to `table`, so I just update the table document.

**Comment by JeremyWuuuuu** [comment]
> > I wonder why is this have to do with `table` documentations, did you somehow mixed your branches together?
> 
> Some props are silimar to `table`, so I just update the table document.

I think we should separate them in order to have better commit tracking, you can create another PR for this.

**Comment by kooriookami** [comment]
> > > I wonder why is this have to do with `table` documentations, did you somehow mixed your branches together?
> > 
> > 
> > Some props are silimar to `table`, so I just update the table document.
> 
> I think we should separate them in order to have better commit tracking, you can create another PR for this.

OK

**Review by jw-foss — APPROVED** [review]

**Comment by kooriookami** [comment]
@HerringtonDarkholme Could you review again?

**Review by HerringtonDarkholme — APPROVED** [review]

**Comment by HerringtonDarkholme** [comment]
We have conflicts! Would you please resolve them?

**Pull request merged**